TWLight/users/models.py

```diff
@@ -156,35 +157,44 @@ class Meta:
     occupation = models.CharField(max_length=128, blank=True)
     affiliation = models.CharField(max_length=128, blank=True)
 
+    def encode_wp_username(self, username):
+        result = urllib.quote(username)
+        return result
+
     @cached_property
     def wp_user_page_url(self):
-        url = u'{base_url}/User:{self.wp_username}'.format(
+        encoded_username = self.encode_wp_username(self.wp_username)
+        url = u'{base_url}/User:{encoded_username}'.format(
             base_url=settings.TWLIGHT_OAUTH_PROVIDER_URL, self=self)
```

Comment:
Because you replaced `{self.wp_username}` with `{encoded_username}` but did not update the dictionary (`.format`), the code is looking for a key (`{encoded_username}`) that's not available in the dictionary. Consider replacing `self` with `encoded_username`.

Reply:
Oh, my mistake, thanks for this feedback @uyscuti-wiki, Let jump on this